### PR TITLE
Improve the code coverage

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log
- add the additional testing method.
- improve the code coverage.(see [result](https://coveralls.io/github/peter279k/uri))

### CODE REVIEW

File: test/UriTest.php [hunk 1]
```diff
@@ -222,10 +222,106 @@ public function testUri($rawUri, $expectedVals) {
...
+        $uri = new Uri('http://www.google.com.tw:54467');
+        $expected = $uri->getAuthority();
+        $this->assertEquals($expected, 'www.google.com.tw:54467');
+    }
+
+    public function testGetAuthorityWithPassword() {
+        $validUri = 'ssh://username:password@example.com:123';
+        $uri = new Uri($validUri);
+        $password = 'password';
+        $uri->getAuthority();
```
Comment (kelunik) [inline test/UriTest.php]: Unnecessary call.

File: test/UriTest.php [hunk 2]
```diff
@@ -222,10 +222,106 @@ public function testUri($rawUri, $expectedVals) {
...
+        $this->assertEquals($password, $uri->getPass());
+    }
+
+    public function testIsIpV4() {
+        $uri = new Uri('http://www.google.com.tw');
+        $expected = $uri->isIpV4();
+        $this->assertFalse($expected);
```
Comment (kelunik) [inline test/UriTest.php]: Would be good to have another test that checks with an IP.

File: test/UriTest.php [hunk 3]
```diff
@@ -222,10 +222,113 @@ public function testUri($rawUri, $expectedVals) {
         $this->assertEquals($expectedVals['fragment'], $uri->getFragment());
     }
 
+    public function testNormalize() {
+        $uri = new Uri('');
+        $this->assertEquals('', $uri->normalize());
+        $normalUri = 'http://google.com.tw/';
+        $uri = new Uri($normalUri);
+        $this->assertEquals($normalUri, $uri->normalize());
+    }
```
Comment (kelunik) [inline test/UriTest.php]: These should probably be two test cases, `testNormalizeEmptyUri` and `testNormalizeBasicUri` or something like that.

File: test/UriTest.php [hunk 4]
```diff
@@ -222,10 +222,113 @@ public function testUri($rawUri, $expectedVals) {
...
         $uri = new Uri('http://localhost/test.php?params=1&params=2');
         $expected = ['params' => [1, 2]];
         $actual = $uri->getAllQueryParameters();
         $this->assertEquals($expected, $actual);
     }
+
+    public function testGetAuthority() {
+        $uri = new Uri('http://www.google.com.tw:54467');
+        $expected = $uri->getAuthority();
+        $this->assertEquals($expected, 'www.google.com.tw:54467');
```
Comment (kelunik) [inline test/UriTest.php]: `'www.google.com.tw:54467'` is the expected value, `$uri->getAuthority()` is the actual value, so the arguments must be switched.

File: test/UriTest.php [hunk 5]
```diff
@@ -222,10 +222,113 @@ public function testUri($rawUri, $expectedVals) {
...
+    }
+
+    public function testGetAuthorityWithPassword() {
+        $validUri = 'ssh://username:password@example.com:123';
+        $uri = new Uri($validUri);
+        $password = 'password';
+        $this->assertEquals($password, $uri->getPass());
```
Comment (kelunik) [inline test/UriTest.php]: You can just use `$this->assertEquals('password', $uri->getPass());` here, no need for an extra variable.

File: test/UriTest.php [hunk 6]
```diff
@@ -222,10 +222,113 @@ public function testUri($rawUri, $expectedVals) {
...
+    }
+
+    public function testGetAuthorityWithHiddenPassword() {
+        $validUri = 'ssh://username:password@example.com:123';
+        $uri = new Uri($validUri);
+        $password = 'password';
+        $uri->getAuthority(true);
```
Comment (kelunik) [inline test/UriTest.php]: Useless method call.

File: test/UriTest.php [hunk 7]
```diff
@@ -222,10 +222,113 @@ public function testUri($rawUri, $expectedVals) {
...
+    public function testGetAuthorityWithPassword() {
+        $validUri = 'ssh://username:password@example.com:123';
+        $uri = new Uri($validUri);
+        $password = 'password';
+        $this->assertEquals($password, $uri->getPass());
+    }
+
+    public function testGetAuthorityWithHiddenPassword() {
+        $validUri = 'ssh://username:password@example.com:123';
+        $uri = new Uri($validUri);
```
Comment (kelunik) [inline test/UriTest.php]: I guess you can just write `new Uri('ssh://...')` here instead of adding another variable.

File: test/UriTest.php [hunk 8]
```diff
@@ -222,10 +222,113 @@ public function testUri($rawUri, $expectedVals) {
...
+        $password = 'password';
+        $uri->getAuthority(true);
+        $this->assertEquals($password, $uri->getPass());
+    }
+
+    public function testIsIpV4() {
+        $uri = new Uri('http://127.0.0.1');
+        $expected = $uri->isIpV4();
+        $this->assertTrue($expected);
```
Comment (kelunik) [inline test/UriTest.php]: Better use `$this->assertTrue($uri->isIpV4())` here, makes the expectation easier to read. Same applies to quite a few other places.

File: test/UriTest.php [hunk 9]
```diff
@@ -222,10 +222,113 @@ public function testUri($rawUri, $expectedVals) {
...
+        $expected = $uri->getQueryParameter('parameter');
+        $this->assertEquals($expected, 'value');
+        $expected = $uri->getQueryParameter('no');
+        $this->assertNull($expected);
+    }
+
+    public function testGetQueryParameterArray() {
+        $uri = new Uri('http://www.google.com.tw?parameter=value');
+        $expected = $uri->getQueryParameterArray('parameter');
+        $this->assertEquals($expected[0], 'value');
```
Comment (kelunik) [inline test/UriTest.php]: This should test for `$this->assertSame(['value'], $uri->getQueryParameterArray('parameter'));`.

File: test/UriTest.php [hunk 10]
```diff
@@ -222,10 +222,113 @@ public function testUri($rawUri, $expectedVals) {
...
+    public function testIsValidDnsName() {
+        $expected = isValidDnsName('google.com');
+        $this->assertTrue($expected);
+        $expected = isValidDnsName('google.com.');
+        $this->assertFalse($expected);
+    }
+
+    public function testInvalidUriException() {
+        $this->expectException(InvalidUriException::class);
+        $uri = new Uri('http://:80');
```
Comment (kelunik) [inline test/UriTest.php]: Variable assignment can be dropped and replaced with just `new Uri(...)`.

File: test/UriTest.php [hunk 11]
```diff
@@ -222,10 +222,113 @@ public function testUri($rawUri, $expectedVals) {
...
+    }
+
+    public function testGetOriginalUri() {
+        $uri = new Uri('http://google.com.tw');
+        $expected = $uri->getOriginalUri();
+        $this->assertEquals($expected, 'http://google.com.tw');
```
Comment (kelunik) [inline test/UriTest.php]: Please test something that gets otherwise normalized, such as `http://google.com:80/../test`.

File: test/UriTest.php [hunk 12]
```diff
@@ -222,10 +222,113 @@ public function testUri($rawUri, $expectedVals) {
...
+    }
+
+    public function testIsValid() {
+        $validUri = 'http://google.com.tw';
+        $inValidUri = 'http://:80';
+        $uri = new Uri($validUri);
+        $expected = $uri->isValid($validUri);
+        $this->assertTrue($expected);
+        $expected = $uri->isValid($inValidUri);
```
Comment (kelunik) [inline test/UriTest.php]: These should be static method calls, so `Uri::isValid()`.

**Review by kelunik — CHANGES_REQUESTED** [review]

**Comment by peter279k** [comment]
Hi @kelunik, thank you for your reply.
I have modified the correct tests that you suggests.
Please review the latest commit. Thanks.

File: test/UriTest.php [hunk 13]
```diff
@@ -222,10 +222,86 @@ public function testUri($rawUri, $expectedVals) {
...
+
+    public function testGetOriginalUri() {
+        $uri = new Uri('http://google.com.tw/../test');
+        $this->assertEquals('http://google.com.tw/../test', $uri->getOriginalUri());
+    }
+
+    public function testIsValid() {
+        $uri = new Uri('http://google.com.tw');
+        $this->assertTrue($uri::isValid('http://google.com.tw'));
+        $this->assertFalse($uri::isValid('http://:80'));
```
Comment (kelunik) [inline test/UriTest.php]: Don't use an instance here, just call the method entirely statically: `Uri::isValid()`.

File: test/UriTest.php [hunk 14]
```diff
@@ -222,10 +222,86 @@ public function testUri($rawUri, $expectedVals) {
...
+    }
+
+    public function testGetAuthority() {
+        $uri = new Uri('http://www.google.com.tw:54467');
+        $this->assertEquals('www.google.com.tw:54467', $uri->getAuthority());
+    }
+
+    public function testGetAuthorityWithPassword() {
+        $uri = new Uri('ssh://username:password@example.com:123');
+        $this->assertEquals('password', $uri->getPass());
```
Comment (kelunik) [inline test/UriTest.php]: Please use `assertSame` over `assertEquals` where possible.

**Review by kelunik — CHANGES_REQUESTED** [review]
Almost good to go, just some minor changes left. :+1:

**Comment by kelunik** [comment]
Awesome, thanks! \o/

**Pull request merged**